smart-contracts/tasks/deploy/set-earliest-block.ts

```diff
@@ -0,0 +1,34 @@
+import networks from '@relay-protocol/networks'
+import { readFileSync, writeFile, writeFileSync } from 'fs'
+import { task } from 'hardhat/config'
+
+const EARLIEST_BLOCKS_FILE =
+  __dirname + '/../../../packages/networks/src/earliestBlocks.json'
+
+function roundDownToThousand(num: number): number {
+  return Math.floor(num / 1000) * 1000
+}
+
+task(
+  'set-earliest-block',
+  'Set the earliest block for a network, if not set yet!'
+).setAction(async (_, { ethers }) => {
+  const { chainId } = await ethers.provider.getNetwork()
+  const network = networks[chainId.toString()]
+  if (!network.earliestBlock) {
+    const data = readFileSync(EARLIEST_BLOCKS_FILE, 'utf-8')
```

Comment:
fwiw I usually use `fs-extra` that has `readJSON` and `outPutJSON` methods. Its already in the repo https://github.com/relayprotocol/relay-vaults/blob/e3ffe60861945e9d41aa055e85ac5abee2a6f8ab/packages/addresses/bin/generateAddressFile.ts#L62

Reply:
ha, good! Let me check this as well!